pkg/controller/rds/dbinstance/setup.go

```diff
@@ -104,6 +107,11 @@ type custom struct {
 	kube     client.Client
 	client   svcsdkapi.RDSAPI
 	external *external
+
+	cache struct {
```

Comment:
Have we done this anywhere else? I wonder if we have any guarantees that this will not be reused for other resources.

I think it is Ok..

Reply:
Yes, I have this multiple times. Since the hook is created for every reconcile, we can be sure that there is always a unique `hook` (or `external`) for every reconcile.